core/services/chainlink/application.go

```diff
@@ -198,6 +201,14 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 	restrictedHTTPClient := opts.RestrictedHTTPClient
 	unrestrictedHTTPClient := opts.UnrestrictedHTTPClient
 
+	// Configure and optionally start the audit log forwarder service
+	auditLogger, err := audit.NewAuditLogger(globalLogger, cfg.AuditLoggerConfig())
+	if err != nil {
+		globalLogger.Info("Audit logger could not be started. err: ", err)
```

Comment:
Should this be a fatal error? Generally if a service cannot be created (let alone started) then something is very wrong that needs to be fixed. I think it would make sense to return a wrapped error here, like pyroscope and nurse do below.

Reply:
This should not be a fatal error because it also occurs when there is no configuration which is probably most of the time.

To stay generic, the AuditLogger takes a config and errors if it cannot create one successfully, in this case though making it a fatal error would mean the logger needs to always be configured

Reply:
Oh, that is surprising and non-standard for both Go and our own internal style. We would prefer that errors stay reserved for true error conditions. Normally we would first do a config check to see if something is enabled before attempting to create it.

Reply:
How about something like this?
```go
if cfg.AuditLoggerEnabled() {
	auditLogger, err = audit.NewAuditLogger(globalLogger, cfg.AuditLoggerConfig())
	if err != nil {
		return nil, errors.Wrap(err, "failed to start Audit Logger")
	}
} else {
	globalLogger.Info("Audit Logger is disabled")
	auditLogger = audit.NoopAuditLogger()
}
```

Reply:
This is pretty much the new system but instead of implementing a NoopAuditLogger which would also need to implement service.ServiceCtx we return an empty AuditLoggerService{} which also does nothing.

Reply:
> instead of implementing a NoopAuditLogger which would also need to implement service.ServiceCtx 

I am not proposing this.

You can still use a zero value `AugitLoggerService`. But for readability, forward compatibility, deduplication, etc. you should define a func to return it. Especially for the callers outside the package or in tests.
```go
package audit

func NoopAuditLogger() { return &AuditLoggerService{} }
```
In fact, `AuditLoggerService` should be unexported as `auditLoggerService` if possible.